src/globus_sdk/services/auth/client/service_client.py

```diff
@@ -348,35 +350,36 @@ def get_identities(
 
         log.debug("Looking up Globus Auth Identities")
 
-        if query_params is None:
-            query_params = {}
-
-        # if either of these params has a truthy value, stringify it
-        if usernames:
-            query_params["usernames"] = commajoin(usernames)
-            query_params["provision"] = (
-                "false" if str(provision).lower() == "false" else "true"
-            )
-        if ids:
-            query_params["ids"] = commajoin(ids)
-
-        log.debug(f"query_params={query_params}")
+        query_params = {
+            "usernames": commajoin(usernames),
+            "ids": commajoin(ids),
+            # only specify `provision` if `usernames` is given
+            "provision": (
+                str(provision).lower() if usernames is not MISSING else MISSING
+            ),
+            **(query_params or {}),
+        }
 
-        if "usernames" in query_params and "ids" in query_params:
+        if (
+            query_params["usernames"] is not MISSING
+            and query_params["ids"] is not MISSING
+        ):
             log.warning(
                 "get_identities call with both usernames and "
                 "identities set! Expected to result in errors"
             )
```

Comment:
The tenses here don't read like a warning to me. Perhaps:

```suggestion
            log.warning(
                "get_identities called with both usernames and "
                "identities set! Expecting an error."
            )
```